tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -261,19 +285,55 @@ static void test_fmt_u64_hex(void)
     uint64_t val = 1002843385516306400;
 
     out[16] = '\0';
+
+    /* Check return count with null buffer input */
+    TEST_ASSERT_EQUAL_INT(16, fmt_u64_hex(NULL, val));
+    /* always 16 regardless of value size */
+    TEST_ASSERT_EQUAL_INT(16, fmt_u64_hex(NULL, 0));
+
     TEST_ASSERT_EQUAL_INT(16, fmt_u64_hex(out, val));
     TEST_ASSERT_EQUAL_STRING("0DEAD0BEEF0CAFE0", (char *) out);
 
     /* check that the buffer was not overflowed */
     TEST_ASSERT_EQUAL_STRING("zz", &out[17]);
 }
 
+static void test_fmt_u16_dec(void)
+{
+    char out[8] = "zzzzzzz";
+    uint16_t val = 6556;
+    uint8_t chars = 0;
+
+    /* Check return count with null buffer input */
+    TEST_ASSERT_EQUAL_INT(5, fmt_u16_dec(NULL, 65535U));
+    TEST_ASSERT_EQUAL_INT(1, fmt_u16_dec(NULL, 0));
+
+    chars = fmt_u16_dec(out, val);
+    TEST_ASSERT_EQUAL_INT(4, chars);
+    out[chars] = '\0';
+    TEST_ASSERT_EQUAL_STRING("6556", (char *) out);
+
+    val = 65535;
+    chars = fmt_u16_dec(out, val);
+    TEST_ASSERT_EQUAL_INT(5, chars);
+    out[chars] = '\0';
+    TEST_ASSERT_EQUAL_STRING("65535", (char *) out);
+
+    /* check that the buffer was not overflowed */
+    TEST_ASSERT_EQUAL_STRING("z", &out[6]);
```

Comment:
Wouldn't you want to check that `out[5]` is not written to either? You could do as in `test_fmt_u16_hex`.

Reply:
No, "65535" is five bytes long. So the buffer should look like this:
```
Adr: 0,1,2,3,4, 5,6, 7
Val: 6,5,5,3,5,\0,z,\0
```
Am I wrong?

Reply:
Yes, but `out[5] = '\0'` only because you overwrite it. You wouldn't notice if the function call wrote out of the buffer by one, right?

Reply:
As discussed in the weekly meeting, the string comparison is replaced with `memcmp`. I believe the test look cleaner know.